### ISSUE TITLE
Expose inverse as transform object

### ISSUE THREAD

**Issue by aeisenbarth** [scikit-image/scikit-image#6921]
It would be great to have a public, short and expressive way to obtain the inverse of a transform as a transform object. After all, if a ProjectiveTransform is invertible, its inverse can be expressed as a matrix and thus also as a ProjectiveTransform. This would help a lot for chaining and composing transforms.

There exists a rather complex introspection mechanism so that the `inverse` method can be passed in place of a transform object, and I have tried understanding the good reasons behind it. But I frequently encounter two obstacles:

- ~~`t1.inverse + t2`~~ (method + ProjectiveTransform) has not been implemented as of https://github.com/scikit-image/scikit-image/pull/1167#issuecomment-56370050
- ~~`t1.inverse.params`~~: If I want to get the projective transform matrix (e.g. to write it to a file), it is not as convenient as it would be for `t1.params`.

## Workaround

```python
ProjectiveTransform(matrix=t1._inv_matrix)
```
The readability of this within a long composition of transforms (with long names) is not so great and I feel not so happy spreading private attribute access in my code.

## Proposal

Add a public property e.g. `inv` which returns a ProjectiveTransform of the inverse (or an appropriate subclass whenever the inverse preserves the original class's constraints).

**Comment by stefanv** [comment]
I'm racking my brain, trying to remember why we did not implement `inverse` on ProjectiveTransform *as* an inverse ProjectiveTransform. I think we can still do that without breaking anything? The only guarantee right now is that `.inverse` takes coordinates and transform them, so returning the inverse ProjectiveTransform would work fine.

**Comment by aeisenbarth** [comment]
I don't know, I was first assuming performance considerations regarding the matrix inversion, but `_inv_matrix` is not cached.

You're right, we don't need introduce a new identifier, since ProjectiveTransform is callable like the `inverse()` method. It should only break code that expects `inverse` to be a method, not a transform.

**Comment by lagru** [comment]
Making `inverse` not return an inverse transform of the former is indeed a bit un-intuitive. I'd propose the following update path

```python
tf = transform.ProjectiveTransform(matrix=....)

tf.inverse(<anything>) 
# Deprecation warning after next release X, otherwise keeps working like now
# Stops accepting input after release X+2

tf.inverse()
# Returns type(tf)(matrix=tf._inv_matrix) after release X
```

This behavior can be achieved by setting a new default signal value like https://github.com/scikit-image/scikit-image/blob/2f9b82ce95e95c558ffe68bced6225c9a4560e31/skimage/_shared/filters.py#L21-L22

If we want, we can then turn `inverse` into a property in skimage2. Though it would be difficult to to so earlier. I can't think of a way for class attribute to behave as a property and and a method at the same time which would be required for a seamless update path.

What do you think?

**Comment by aeisenbarth** [comment]
Yes, so we agree on the approach converting `inverse` to a property which returns the inverse transform object?

**Comment by stefanv** [comment]
Let's try it and see whether we run into any problems.

**Comment by aeisenbarth** [comment]
Closing this because it is resolved by PR #6926 and released with `scikit-image==0.21`.
